### PR TITLE
Adds PATCH /api/articles/:article_id endpoint with error handling and associated tests

### PR DESCRIPTION
**Pull request by Radishetta**
Creates PATCH /api/articles/article_id endpoint
Updates app.test.js
Updates app.js
Updates articles.controller.js
Updates articles.model.js
Updates errors/index.js
Updates endpoints.json

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -254,7 +254,56 @@ describe("POST /api/articles/:article_id/comments", () => {
...
+          title: expect.any(String),
+          topic: "mitch",
+          author: "butter_bridge",
+          body: expect.any(String),
+          created_at: expect.any(String),
+          votes: 110,
+          article_img_url: expect.any(String),
+        });
+      });
+  });
```
Comment (MKD01) [inline __tests__/app.test.js:284]: Really good first test.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -254,7 +254,56 @@ describe("POST /api/articles/:article_id/comments", () => {
...
+
+  it("400: ERROR - responds with an error message when passed an object with invalid data type", () => {
+    const newUpdate = { inc_votes: "imAVote" };
+
+    return request(app)
+      .patch("/api/articles/1")
+      .send(newUpdate)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
```
Comment (MKD01) [inline __tests__/app.test.js:306]: Really good error handling, but dont forget that we also need to include error handling for when we are given an invalid article id and a valid article id but one that does not exist.

**Review by MKD01 — COMMENTED** [review]
Really good work but there are a couple of missing tests. Once those have been added feel free to merge them in without needing to send us the updated pr.

**Comment by Radishetta** [comment]
Thanks a lot for the feedback, you are absolutely right. Since its a new endpoint I have to test again for article_id

**Pull request merged**